### ISSUE TITLE
port should be optional

### ISSUE THREAD

**Issue by dpastoor** [typesense/typesense-js#12]
Especially if typesense is behind a reverse proxy, the url does not necessarily have a url exposed.

An example setup:

```
https://example.com/typesense
```

proxies internally to the typesense docker container running at localhost:8108

**Comment by kishorenc** [comment]
@dpastoor You can define the port to be `443`, which is the default HTTPS port. So, `https://github.com` and `https://github.com:443` are equivalent. Let me know if that works!

**Comment by dpastoor** [comment]
To be clear, the problem is when the host is NON root.

Yes, `https://github.com` and `https://github.com:443`

but `https://github.com/typesense:443` is **not valid**. 

The way the code is constructed:

```
`${this._configuration.masterNode.protocol}://${this._configuration.masterNode.host}:${this._configuration.masterNode.port}${endpoint}`
```

makes it such non-root urls to typesense, such as `example.com/typesense` impossible to use as the endpoint right now.

but if typesense is served at example.com/typesense

**Comment by kishorenc** [comment]
@dpastoor Thank you for the clarification. I can now see how this is a potential blocker. It definitely makes reverse-proxying of Typesense on the same host as an app difficult. 

We will address this issue and release a new version of the library by next week.

**Comment by dpastoor** [comment]
Fixed in #14 and should be easy to adapt to the other libraries.
